src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/service/impl/IdamElasticSearchServiceImpl.java

```diff
@@ -220,50 +218,51 @@ public ResponseEntity<Object> getIdamDetails() {
             RefDataElinksConstants.JobStatus.IN_PROGRESS.getStatus(), IDAMSEARCH);
         Set<IdamResponse> judicialUsers = new HashSet<>();
         List<UserProfile> userProfiles = userProfileRepository.fetchObjectIdFromCurrentDate();
-        userProfiles.forEach(userProfile -> {
-            Map<String, String> params = new HashMap<>();
-            params.put("query", idamFindQuery.concat(userProfile.getObjectId()));
-            log.debug("{}:: search elk query {}", loggingComponentName, params.get("query"));
-            int count = 0;
-            int totalCount = 0;
-            ResponseEntity<Object> responseEntity = null;
-            Response response = null;
-            try {
-                String bearerToken = "Bearer ".concat(getIdamBearerToken(schedulerStartTime));
-                response = idamFeignClient.getUserFeed(bearerToken, params);
-                logIdamResponses(response);
-                responseEntity = JsonFeignResponseUtil.toResponseEntity(response,
-                    new TypeReference<Set<IdamResponse>>() {
-                    });
-                if (response.status() == 200) {
-
-                    Set<IdamResponse> users = (Set<IdamResponse>) responseEntity.getBody();
-                    judicialUsers.addAll(users);
-
-                    List<String> headerCount = responseEntity.getHeaders().get("X-Total-Count");
-                    if (headerCount != null && !headerCount.isEmpty()
-                        && !headerCount.get(0).isEmpty()) {
-
-                        totalCount = Integer.parseInt(headerCount.get(0));
-                        log.debug("{}:: Header Records count from Idam :: " + totalCount, loggingComponentName);
+        if (userProfiles.size() > 0) {
```

Comment:
Would be good to log the count before this condition so you can tell when the condition will get skipped

Reply:
logged count